command_test.go

```diff
@@ -316,6 +316,7 @@ func TestCancel(t *testing.T) {
 	}
 
 	re, res := NewChanResponsePair(req)
+	cancel()
```

Comment:
>`cancel` should likely be called before the `Emit` goroutine even has the potential to run.

Incorrect. This needs to go back where it was.
We pass either way, but at least when `cancel` is called late, there's a 50/50 chance we also test that pending sends will be canceled instead of sent.
Depending on who runs first, `Emit` or `cancel`.